File: RQ3/Java-REval-Desc/task84_desc.java
public class Solution {
    public static String solve(String s) {
        int flg = 0;
        int idx = 0;
        char[] new_str = s.toCharArray();
        for (char i : s.toCharArray()) {
            if (Character.isAlphabetic(i)) {
                new_str[idx] = Character.isLowerCase(i) ? Character.toUpperCase(i) : Character.toLowerCase(i);
                flg = 1;
            }
            idx++;
        }
        s = new String(new_str);
        if (flg == 0) {
            StringBuilder _result = new StringBuilder(s).reverse();
        } else {
            String _result = s;
        }

        assert func(_result);  // POST: func is equivalent to: You are given a string s. if s[i] is a letter, reverse its case otherwise keep it as it is. If the string contains no letters, reverse the string. The function should return the resulted string. Examples solve("1234") = "4321" solve("ab") = "AB" solve("#a@C") = "#A@c"
    }
}

File: RQ3/C-REval-Desc/task84_desc.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

void solve(char *s) {
    int flg = 0;
    int idx = 0;
    char new_str[1000];
    strcpy(new_str, s);
    int len = strlen(s);

    for (int i = 0; i < len; i++) {
        if (isalpha(s[i])) {
            new_str[idx] = islower(s[i]) ? toupper(s[i]) : tolower(s[i]);
            flg = 1;
            idx++;
        } else {
            new_str[idx] = s[i];
            idx++;
        }
    }
    new_str[idx] = '\0';

    if (flg == 0) {
        char _result[1000];
        for (int i = 0; i < len; i++) {
            _result[i] = new_str[len - 1 - i];
        }
        _result[len] = '\0';
        assert(func(_result));
    } else {
        assert(func(new_str));
    }
}